### PR TITLE
BZ#1858125 - Include network policy rule optimization

### PR DESCRIPTION
**Pull request by jboxman**
- https://bugzilla.redhat.com/show_bug.cgi?id=1858125

### CODE REVIEW

**Comment by jboxman** [comment]
Hi @juanluisvaladas,

I've migrated this from the OCP 3.11 docs; Is it still accurate?

Thanks!

File: modules/nw-networkpolicy-optimize.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+====
+
+It is inefficient to apply `NetworkPolicy` objects to large numbers of individual pods in a single namespace. Pod labels do not exist at the IP address level, so network policy generates a separate Open vSwitch (OVS) flow rule for every single possible link between every pod selected with a `podSelector`.
+
+For example, if the spec `podSelector` and the ingress `podSelector` within a `NetworkPolicy` object each match 200 pods, then 40,000 (200*200) OVS flow rules are generated. This might slow down a node.
+
+When designing your network policy, refer to the following guidelines:
+
+* Reduce the number of OVS flow rules by using namespaces to contain groups of pods that need to be isolated.
++
```
Comment (danwinship) [inline modules/nw-networkpolicy-optimize.adoc:22]: (looks like you accidentally pasted a `+` in there)

**Review by danwinship — COMMENTED** [review]
Yes, this still applies to openshift-sdn in 4.x.

File: modules/nw-networkpolicy-optimize.adoc (hunk 1 above)
Comment (jboxman) [reply modules/nw-networkpolicy-optimize.adoc:22]: @danwinship, oops, thanks!

File: modules/nw-networkpolicy-optimize.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Module included in the following assemblies:
+//
+// * networking/network_policy/about-network-policy.adoc
+
+[id="nw-networkpolicy-optimize_{context}"]
+= Optimizations for network policy
+
+A network policy allows you to isolate pods that are differentiated from one another by labels within a namespace.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-optimize.adoc:8]:
```suggestion
Use a network policy to isolate pods that are differentiated from one another by labels within a namespace.
```

File: modules/nw-networkpolicy-optimize.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+[NOTE]
+====
+The guidelines for efficient use of network policy rules applies to only the OpenShift SDN cluster network provider.
+====
+
+It is inefficient to apply `NetworkPolicy` objects to large numbers of individual pods in a single namespace. Pod labels do not exist at the IP address level, so network policy generates a separate Open vSwitch (OVS) flow rule for every single possible link between every pod selected with a `podSelector`.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-optimize.adoc:15]:
```suggestion
It is inefficient to apply `NetworkPolicy` objects to large numbers of individual pods in a single namespace. Pod labels do not exist at the IP address level, so a network policy generates a separate Open vSwitch (OVS) flow rule for every possible link between every pod selected with a `podSelector`.
```

File: modules/nw-networkpolicy-optimize.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+For example, if the spec `podSelector` and the ingress `podSelector` within a `NetworkPolicy` object each match 200 pods, then 40,000 (200*200) OVS flow rules are generated. This might slow down a node.
+
+When designing your network policy, refer to the following guidelines:
+
+* Reduce the number of OVS flow rules by using namespaces to contain groups of pods that need to be isolated.
++
+`NetworkPolicy` objects that select a whole namespace, by using the `namespaceSelector` or an empty `podSelector`, only generate a single OVS flow rule that matches the VXLAN virtual network ID (VNID) of the namespace.
```
Comment (kalexand-rh) [inline modules/nw-networkpolicy-optimize.adoc:23]:
```suggestion
`NetworkPolicy` objects that select a whole namespace, by using the `namespaceSelector` or an empty `podSelector`, generate only a single OVS flow rule that matches the VXLAN virtual network ID (VNID) of the namespace.
```

**Review by kalexand-rh — COMMENTED** [review]
This is looking good!

**Comment by jboxman** [comment]
@kalexand-rh thanks!

**Pull request merged**